lib/class-name.js

```diff
@@ -0,0 +1,25 @@
+"use strict";
+
+var functionName = require("./function-name");
+
+module.exports = function className(value) {
+    if (value.constructor && "name" in value.constructor) {
+        return value.constructor.name;
+    }
+
+    // The next branch is for IE11 support only:
+    // Because the name property is not set on the prototype
+    // of the Function object, we finally try to grab the
+    // name from its definition. This will never be reached
+    // in node, so we are not able to test this properly.
+    // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name
+    if (typeof value.constructor === "function") {
+        var name = functionName(value.constructor);
+
+        if (name) {
+            return name;
+        }
+    }
+
+    return null;
```

Comment:
I think this could be simplified drastically:

```js
return value.constructor && value.constructor.name
  || typeof value.constructor === 'function' && functionName(value.constructor)
  || null
;
```

Reply:
@fearphage I totally like your version. Thanks a lot for your contribution, I will update the branch accordingly

Reply:
Changed

Reply:
BTW, I left the comment above the affecting branch